packages/relay/src/lib/services/debugService/index.ts

```diff
@@ -0,0 +1,261 @@
+   */
+  constructor(mirrorNodeClient: MirrorNodeClient, logger: Logger, common: CommonService) {
+    this.logger = logger;
+    this.common = common;
+    this.mirrorNodeClient = mirrorNodeClient;
+  }
+
+  /**
+   * Checks if the Debug API is enabled
+   * @public
+   */
+  public static readonly isDebugAPIEnabled = process.env.DEBUG_API_ENABLED;
+
+  static requireDebugAPIEnabled(): void {
+    if (!process.env.DEBUG_API_ENABLED || process.env.DEBUG_API_ENABLED !== 'true') {
+      throw predefined.UNSUPPORTED_METHOD;
+    }
+  }
+
+  /**
+   * Trace a transaction for debugging purposes.
+   *
+   * @async
+   * @param {string} transactionHash - The hash of the transaction to be traced.
+   * @param {TracerType} tracer - The type of tracer to use (either 'CallTracer' or 'OpcodeLogger').
+   * @param {object} tracerConfig - The configuration object for the tracer.
+   * @param {string} [requestIdPrefix] - An optional request id.
+   * @throws {Error} Throws an error if the specified tracer type is not supported or if an exception occurs during the trace.
+   * @returns {Promise<any>} A Promise that resolves to the result of the trace operation.
+   *
+   * @example
+   * const result = await debug_traceTransaction('0x123abc', TracerType.CallTracer, {"tracerConfig": {"onlyTopCall": false}}, some request id);
+   */
+  async debug_traceTransaction(
+    transactionHash: string,
+    tracer: TracerType,
+    tracerConfig: object,
+    requestIdPrefix?: string,
+  ): Promise<any> {
+    this.logger.trace(`${requestIdPrefix} debug_traceTransaction(${transactionHash})`);
+    try {
+      DebugService.requireDebugAPIEnabled();
+      if (tracer === TracerType.CallTracer) {
+        return await this.callTracer(transactionHash, tracerConfig, requestIdPrefix);
+      } else if (tracer === TracerType.OpcodeLogger) {
+        throw predefined.UNSUPPORTED_METHOD;
+      }
+    } catch (e) {
+      throw this.common.genericErrorHandler(e);
+    }
+  }
+
+  /**
+   * Formats the result from the actions endpoint to the expected response
+   *
+   * @async
+   * @param {any} result - The response from the actions endpoint.
+   * @param {string} requestIdPrefix - The request prefix id.
+   * @returns {Promise<[] | any>} The formatted actions response in an array.
+   */
+  async formatActionsResult(result: any, requestIdPrefix?: string): Promise<[] | any> {
+    return await Promise.all(
+      result.map(async (action, index) => {
+        const { resolvedFrom, resolvedTo } = await this.resolveMultipleAddresses(
+          action.from,
+          action.to,
+          requestIdPrefix,
+        );
+
+        // The actions endpoint does not return input and output for the calls so we get them from another endpoint
+        // The first one is excluded because we take its input and output from the contracts/results/{transactionIdOrHash} endpoint
+        const getContract =
+          index !== 0 && action.call_operation_type === 'CREATE'
```

Comment:
would it be valuable to create an Enum with the possible values of this `action.call_operation_type` and compare against the Enum?

Reply:
Well, posibly yes, I will add it